config/v1/types_oauth.go

```diff
@@ -529,22 +521,38 @@ const UserIDClaim = "sub"
 // OpenIDClaims contains a list of OpenID claims to use when authenticating with an OpenID identity provider
 type OpenIDClaims struct {
 	// preferredUsername is the list of claims whose values should be used as the preferred username.
-	// If unspecified, the preferred username is determined from the value of the id claim
+	// If unspecified, the preferred username is determined from the value of the sub claim
 	// +optional
 	PreferredUsername []string `json:"preferredUsername"`
+
 	// name is the list of claims whose values should be used as the display name. Optional.
 	// If unspecified, no display name is set for the identity
 	// +optional
 	Name []string `json:"name"`
+
 	// email is the list of claims whose values should be used as the email address. Optional.
 	// If unspecified, no email is set for the identity
 	// +optional
 	Email []string `json:"email"`
 }
 
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
 type OAuthList struct {
 	metav1.TypeMeta `json:",inline"`
 	metav1.ListMeta `json:"metadata,omitempty"`
-	Items           []OAuth `json:"items"`
+
+	Items []OAuth `json:"items"`
+}
+
+// OAuthConfigMapReference references a config map in the openshift-config namespace.
+type OAuthConfigMapReference struct {
+	// name is the metadata.name of the referenced config map
+	Name string `json:"name"`
+}
+
+// OAuthSecretReference references a secret in the openshift-config namespace.
+type OAuthSecretReference struct {
```

Comment:
These refs are generic.  Name them as such.  You can rename the others as `SecretFileRef` or some such.